src/argilla/client/feedback/training/frameworks/trl.py

```diff
@@ -0,0 +1,383 @@
+from argilla.client.feedback.training.schemas import (
+    TrainingTaskForDPO,
+    TrainingTaskForPPO,
+    TrainingTaskForRM,
+    TrainingTaskForSFT,
+)
+from argilla.training.utils import filter_allowed_args
+from argilla.utils.dependency import require_version
+
+if TYPE_CHECKING:
+    import transformers
+    from trl import PPOConfig
+
+    from argilla.client.feedback.dataset import FeedbackDataset
+
+
+class PPOArgs:
+    def __init__(
+        self,
+        config: "PPOConfig",
+        reward_model: Union[str, "transformers.pipeline"],
+        length_sampler_kwargs: dict,
+        generation_kwargs: dict,
+    ) -> None:
+        """
+        Additional arguments for PPO training process.
+
+        Args:
+            reward_model (Union[str, "transformers.pipeline"]): Reward model to use for creating PPO rewards for training.
+            length_sampler_kwargs (dict): Arguments for the length sampler.
+                min_value: Minimum length for the generated samples.
+                max_value: Maximum length for the generated samples.
+            generation_kwargs (dict): Arguments for the generation process.
+                min_length: Minimum length for the generated samples.
+                max_length: Maximum length for the generated samples.
+                num_beams: Number of beams to use for the generation process.
+                num_return_sequences: Number of sequences to generate.
+                temperature: Temperature for the generation process.
+                top_k: Top k for the generation process.
+                top_p: Top p for the generation process.
+        """
+        if isinstance(reward_model, str):
+            from transformers import pipeline
+
+            reward_model = pipeline("text-classifcation", model=reward_model)
+        self.config = config
+        self.reward_model = reward_model
+        self.length_sampler_kwargs = length_sampler_kwargs
+        self.generation_kwargs = generation_kwargs
+
+
+class ArgillaTRLTrainer(ArgillaTrainerSkeleton):
+    _logger = logging.getLogger("ArgillaTRLTrainer")
+    _logger.setLevel(logging.INFO)
+
+    require_version("trl>=0.5.0")
+
+    def __init__(
+        self,
+        dataset: "FeedbackDataset",
+        task: Union[
+            TrainingTaskForSFT,
+            TrainingTaskForRM,
+            TrainingTaskForPPO,
+            TrainingTaskForDPO,
+        ],
+        prepared_data=None,
+        model: str = None,
+        seed: int = None,
+    ):
+        super().__init__(dataset=dataset, task=task, prepared_data=prepared_data, model=model, seed=seed)
+        import torch
+        from datasets import DatasetDict
+        from transformers import set_seed
+
+        self._transformers_model = None
+        self._transformers_tokenizer = None
+        self.device = "cpu"
+        if torch.backends.mps.is_available():
+            self.device = "mps"
+        elif torch.cuda.is_available():
+            self.device = "cuda"
```

Comment:
Could we include a `use_cuda` and `use_mps` boolean flags as args? Maybe we don't want to use the device even if available

Reply:
I agree, this was to keep people from figuring out how to use `cuda` and `mps` but it can be overwritten via `update_config`. Perhaps it would be better to include it as one of the main args like `train_size`, however, this can be tackled in a different PR IMO. I believe @dvsrepo also suggested some other API changes.

Reply:
I think it might be best to let users provide a `device` to the ArgillaTrainer init directly.

Reply:
Let's track this in #3632 after this PR.